### PR TITLE
navigate to latest screen

### PR DESCRIPTION
**Pull request by ltminhthu**

### CODE REVIEW

File: apps/web/src/screens/Dashboard/index.tsx [hunk 1]
```diff
@@ -28,6 +28,8 @@ export const DashboardScreen: FC = () => {
 		await router.navigate('/');
 	};
 
+	localStorage.setItem('pathname', pathname);
+
```
Comment (FME849) [inline apps/web/src/screens/Dashboard/index.tsx:32]: I think this feature could be reusable on mobile, thus we should not use `localStorage` here. You could try to store the layout `id` in setting document in DB and then navigate based on that `id`.

Comment (ltminhthu) [reply apps/web/src/screens/Dashboard/index.tsx:32]: do we need to synchronize the `layout_id` between mobile and extension?

Comment (ltminhthu) [reply apps/web/src/screens/Dashboard/index.tsx:32]: > I think this feature could be reusable on mobile, thus we should not use `localStorage` here. You could try to store the layout `id` in setting document in DB and then navigate based on that `id`.

btw, do we need to store screen like "setting" or "profile"

Comment (FME849) [reply apps/web/src/screens/Dashboard/index.tsx:32]: > do we need to synchronize the `layout_id` between mobile and extension?

do you mean synchronize last project `id` between mobile and extension? I think those platforms share the feature not the data

Comment (FME849) [reply apps/web/src/screens/Dashboard/index.tsx:32]: > > I think this feature could be reusable on mobile, thus we should not use `localStorage` here. You could try to store the layout `id` in setting document in DB and then navigate based on that `id`.
> 
> btw, do we need to store screen like "setting" or "profile"

I can't think of any purpose to save `setting` or `profile` screens, but we could check again with others.

File: apps/web/src/screens/Dashboard/index.tsx [hunk 2]
```diff
@@ -28,6 +29,17 @@ export const DashboardScreen: FC = () => {
...
+		if (pathname !== setting?.currentScreen) {
+			console.log(setting?.currentScreen);
+			await setPathname({ currentScreen: pathname });
+		}
+	};
+
+	useEffect(() => {
+		setLatestPathname();
+	}, [pathname]);
+
```
Comment (FME849) [inline apps/web/src/screens/Dashboard/index.tsx:42]: I think `async/await` is not required here, we can reduce these line like this:
```typescript
  useEffect(() => {
    if (pathname !== setting?.currentScreen) {
     setPathname(pathname);
    }
  }, [pathname]);
```

File: apps/web/src/state/app/settings.ts [hunk 3]
```diff
@@ -18,3 +18,15 @@ export const setPrivacy = async ({ hideBalance }: { hideBalance: boolean }) => {
...
+	currentScreen,
+}: {
+	currentScreen: string;
+}) => {
+	await modules.storage.upsert<SettingDocument>(id, async (doc) => {
+		doc.config.currentScreen = currentScreen;
+
+		return doc;
+	});
+};
```
Comment (FME849) [inline apps/web/src/state/app/settings.ts:32]: I realized that there's a parameter mistake in my previous `setPrivacy` that it does not need to destruct `hideBalance` there. I will explain more in chat.

File: apps/web/src/screens/Dashboard/index.tsx [hunk 4]
```diff
@@ -28,6 +29,17 @@ export const DashboardScreen: FC = () => {
 		await router.navigate('/');
 	};
 
+	const setLatestPathname = async () => {
+		if (pathname !== setting?.currentScreen) {
+			console.log(setting?.currentScreen);
```
Comment (fluctlight-kayaba) [inline apps/web/src/screens/Dashboard/index.tsx:34]: should clean console.log before push

File: apps/web/src/screens/Dashboard/index.tsx [hunk 5]
```diff
@@ -28,6 +29,17 @@ export const DashboardScreen: FC = () => {
...
+			await setPathname({ currentScreen: pathname });
+		}
+	};
+
+	useEffect(() => {
```
Comment (fluctlight-kayaba) [inline apps/web/src/screens/Dashboard/index.tsx:39]: this observe/watch flow should be in higher level, e.g App to ensure it work for other routes as well

File: packages/app/state/app.ts [hunk 6]
```diff
@@ -25,5 +25,6 @@ export const appState = proxy<AppState>({
 	authenticationLoading: false,
 	config: {
 		hideBalance: true,
+		currentScreen: '/',
```
Comment (fluctlight-kayaba) [inline packages/app/state/app.ts:28]: propose use more generic name e.g `latestLocation`

File: packages/core/utils/entityTypes.ts [hunk 7]
```diff
@@ -9,6 +9,7 @@ export interface Setting {
 
 export interface Config {
 	hideBalance: boolean;
+	currentScreen: string;
```
Comment (fluctlight-kayaba) [inline packages/core/utils/entityTypes.ts:12]: consider another name

**Pull request closed without merging**